packages/concerto-core/lib/introspect/scalardeclaration.js

```diff
@@ -278,6 +278,16 @@ class ScalarDeclaration extends Declaration {
         return false;
     }
 
+    /**
+     * Returns true if this class is the definition of a map-declaration.
+     *
+     * @return {boolean} true if the class is a map-declaration
+     * @deprecated
+     */
+    isMapDeclaration() {
```

Comment:
Should this be part of the parent class and then overloaded in `MapDeclaration`?

Reply:
Yep! implemented as you suggested.